genie-core/src/main/java/com/netflix/genie/core/services/impl/JobCoordinatorServiceImpl.java

```diff
@@ -240,6 +241,9 @@ public String coordinateJob(
                     );
                 }
             }
+        } catch (GenieConflictException e) {
```

Comment:
This will call done on an already running job if there is indeed a conflict. As discussed in person we need a better solution for this.